src/aia_clic_extension.adoc

```diff
@@ -0,0 +1,1816 @@
+:inhv: pass:q[``**__x__**inhv``]
+
+:mpil: pass:q[``mpil``]
+:minhv: pass:q[``minhv``]
+
+:ret: pass:q[``**__x__**ret``]
+:mret: pass:q[``mret``]
+
+:le: &#8804;
+:ge: &#8805;
+:lt: &#60;
+:gt: &#62;
+
+// SPDX-License-Identifier: CC-BY-4.0
+//
+// licensing.adoc: licensing information
+//
+// Copyright and licensing information for the specification.
+//
+
+[Preface]
+== Copyright and license information
+
+This RISC-V AIA CLIC extension specification is © 2018-2025 RISC-V international
+
+This document is released under a Creative Commons Attribution 4.0
+International License. +
+https://creativecommons.org/licenses/by/4.0/.
+
+Please cite as: “Core-Local Interrupt Controller (CLIC) RISC-V Privileged Architecture Extensions",  RISC-V International
+
+This document is a derivative of the "The RISC-V Instruction Set Manual, Volume II: Privileged Architecture, Document
+version 1.9.1" released under following license: © 2010–2017 Andrew Waterman,
+Yunsup Lee, Rimas Aviˇzienis, David Patterson, Krste Asanovi ́c.
+Creative Commons Attribution 4.0 International License.
+
+[Preface]
+== Revision History
+
+[source]
+----
+Date        Description
+----
+
+== CLIC extensions to AIA
+This section gives an overview for the Core-Local Interrupt
+Controller (CLIC) extensions to AIA.
+
+This table provides a summary of the CLIC extensions to AIA.
+
+[%autowidth]
+|===
+| Extension Name | Description
+| smclicincr   | Increase of the number of local interrupts for M-mode
+| ssclicincr   | Increase of the number of local interrupts for S-mode
+| smclic       | Same Privilege Mode Interrupt Preemption support for M-mode
+| ssclic       | Same Privilege Mode Interrupt Preemption support for S-mode
+| smclicshv    | Selective Hardware Vectored Interrupts
+| smclicconfig | Allows implementations to support different parameterizations of CLIC extensions
+| smclicdbg    | support for interrupt debug triggering
+| smclicip     | support for interrupt context push/pop
+| smcliclat    | features to improve latency of context switching.
+|===
+
+== Increase of AIA local interrupts - smclicincr
+
+The smclicincr extension increases support up to 4096 interrupt inputs per hart.
+Each interrupt input _i_ has five control
+registers: an interrupt-pending bit (`clicintip[__i__]`),
+an interrupt-enable bit (`clicintie[__i__]`), interrupt attributes
+(`clicintattr[__i__]`) to specify trigger type, 
+(`cliciprio[__i__]`) to specify priority, and 
+(`clicmideleg[__i__]`) to delegate interrupts to a lower privilege level.
+
+NOTE: The existing timer (`mtip`/`stip`), software
+(`msip`/`ssip`), and external interrupt inputs
+(`meip`/`seip`) can be treated as additional local interrupt
+sources, where the privilege mode, interrupt level, and priority can
+be altered using `clicintattr[__i__]` and
+`cliciprio[__i__]` and `clicmideleg[__i__]` registers.
+
+NOTE: To maintain mip behavior, interrupts with read-only behavior should be configured/hard-coded with trigger type level and

```

Comment:
I am not sure if this is a note or a normative requirement.
Also, requiring the edge triggered behavior might be too restrictive.